src/getVulnerabilityByCve.js

```diff
@@ -0,0 +1,49 @@
+const authenticatedRequest = require('./authenticatedRequest');
+const { getLogger } = require('./logger');
+const { RequestError } = require('./responses');
+
+/**
+ * Fetches vulnerability details from CrowdStrike Spotlight API for a given CVE ID.
+ * @param {string} cveId - The CVE ID to filter on (e.g., 'CVE-2023-12345').
+ * @param {object} options - Integration options (auth, etc).
+ * @returns {Promise<object>} - Structured API response data
+ */
+async function getVulnerabilityByCve(cveId, options) {
+  const Logger = getLogger();
+
+  const filter = `cve.id:['${cveId}']`;
+  const url = `${
+    options.url
+  }/spotlight/combined/vulnerabilities/v1?filter=${encodeURIComponent(
```

Comment:
One minor change here is we should use the `qs` parameters on the request to pass query parameters rather than pass them manually.  So the request options would look like this:

```
  const requestOptions = {
    url: options.url,
    qs: {
       filter,
       facet: ['host_info', 'cve', 'remediation', 'evaluation_logic']
    },
    json: true
  };
```